src/main/java/com/fasterxml/jackson/databind/util/NativeImageUtil.java

```diff
@@ -9,31 +9,39 @@
  * @since 2.14
  */
 public class NativeImageUtil {
-    private static final boolean RUNNING_IN_SVM;
-
-    static {
-        RUNNING_IN_SVM = System.getProperty("org.graalvm.nativeimage.imagecode") != null;
-    }
+    
+    private static final boolean RUNNING_IN_SVM = isInSVM();
 
     private NativeImageUtil() {
     }
 
     /**
-     * Check whether we're running in substratevm native image runtime mode.
-     * This check cannot be a constant, because
+     * Check whether we're running in SubstrateVM native image and also in "runtime" mode.
+     * The "runtime" check cannot be a constant, because
      * the static initializer may run early during build time
      *<p>
-     * NOTE: {@code public} since 2.16 (before that, {@code private}).
+     * As optimization, {@link #RUNNING_IN_SVM} is used to short-circuit on normal JVMs.
      */
-    public static boolean isRunningInNativeImage() {
+    private static boolean isInNativeImageAndIsAtRuntime() {
```

Comment:
Hmmh. But is this what Afterburner and Blackbird need? I thought they'd specifically need `isInNativeImageAndIsAtRuntime` and not just `isInNativeImage()`?

Or does it not matter since AB/BB should never be enabled even during build phase?

/cc @yawkat

Reply:
I believe it is the latter based on the original discussion; BB / AB should not be active in either build or run time for SubstrateVM and therefore they should use this newly added `isInNativeImage`.

Reply:
Ok. I'll merge this & make public -- while minimal interface sometimes useful, I think in this case both methods can be useful.